tests/xharness/Logging/CaptureLog.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.IO;
+
+namespace Xharness.Logging {
+
+	// A log that captures data written to a separate file between two moments in time
+	// (between StartCapture and StopCapture).
+	public class CaptureLog : Log
+	{
+		public string CapturePath { get; private set; }
+		public string Path { get; private set; }
+
+		long startPosition;
+		long endPosition;
+		bool entire_file;
+		bool started;
+
+		public CaptureLog (ILogs logs, string path, string capture_path, bool entire_file = false)
+			: base (logs)
+		{
+			if (path == null)
+				throw new ArgumentNullException (nameof (path));
+
+			Path = path;
+			CapturePath = capture_path;
+			this.entire_file = entire_file;
+		}
+
+		public void StartCapture ()
+		{
+			if (entire_file)
+				return;
+
+			if (File.Exists (CapturePath))
+				startPosition = new FileInfo (CapturePath).Length;
+			started = true;
+		}
+
+		public void StopCapture ()
+		{
+			if (!started && !entire_file)
+				throw new InvalidOperationException ("StartCapture most be called before StopCature on when the entire file will be captured.");
+			if (!File.Exists (CapturePath)) {
+				File.WriteAllText (Path, $"Could not capture the file '{CapturePath}' because it doesn't exist.");
+				return;
+			}
+
+			if (entire_file) {
+				File.Copy (CapturePath, Path, true);
+				return;
+			}
+
+			endPosition = new FileInfo (CapturePath).Length;
+
+			Capture ();
+		}
+
+		void Capture ()
+		{
+			if (startPosition == 0 || entire_file)
+				return;
+
+			if (!File.Exists (CapturePath)) {
+				File.WriteAllText (Path, $"Could not capture the file '{CapturePath}' because it does not exist.");
+				return;
+			}
+
+			var currentEndPosition = endPosition;
+			if (currentEndPosition == 0)
```

Comment:
>if (currentEndPosition == 0) [](start = 3, length = 28)

This would happen for empty file only, no? So we can create that empty file and be done here.
Or if this is for entire_file mode, than this is handled already above and shouldn't happen here.

This is example of the impossible state of the class too

Reply:
No, lets imaging we have several task/processes writing to that file (assuming no share violations). If at the point of stop we have an empty file, but when Capture gets called a flush happens, we could be in a situation in which the file has content but our information is outdated. @rolfbjarne is there a reason why you added this?